### PR TITLE
simplify -  fixes issue #21340 and now trigsimp with fu method returns nan for zero denominator

### LINKED ISSUE #21340: `trigsimp` of `(tan(x)**2 - sec(x)**2 + 1) / (sin(x)**2 + cos(x)**2 - 1)`
This is one of the problem mentioned in *Lamagna, Computer Algebra: Concepts and Techniques, Ch.9* that can test out the fallacy of automated cancelling `f(x)/f(x)`. I think that the result of `fu` is incorrect.

```python3
In [1]: x = Symbol('x')

In [2]: expr = (tan(x)**2 - sec(x)**2 + 1) / (sin(x)**2 + cos(x)**2 - 1)

In [3]: simplify(expr)
Out[3]: nan

In [4]: trigsimp(expr)
Out[4]: nan

In [5]: trigsimp(expr, method='fu')
Out[5]:
   1
───────
   2
cos (x)
```

### PR DESCRIPTION
**Pull request by praneethratna**
#### Brief description of what is fixed or changed
Fixes #21340 and now `trigsimp((tan(x)**2 - sec(x)**2 + 1) / (sin(x)**2 + cos(x)**2 - 1), method='fu')` returns `nan` instead of `cos(x)**(-2)`

#### Release Notes

* simplify
   * fixed bug in fu.py

### CODE REVIEW

**Comment by praneethratna** [comment]
@smichr please review

File: sympy/simplify/fu.py [hunk 1]
```diff
@@ -1643,6 +1644,10 @@ def fu(rv, measure=lambda x: (L(x), x.count_ops())):
 
     was = rv
     rv = sympify(rv)
+    if isinstance(rv, Expr):
+        n, d = fraction(rv)
+        if simplify(d) is S.Zero:
```
[the remark this answers is not in the document]
Comment (praneethratna) [reply sympy/simplify/fu.py]: @oscargus I've made the changes.

File: .mailmap [hunk 2]
```diff
@@ -567,7 +568,6 @@ Aman Sharma <amansharma110603@gmail.com> mostlyaman <85072876+mostlyaman@users.n
 Robin Richard <raisin@ecomail.fr> robinechuca <raisin@ecomail.fr>
 Robin Richard <raisin@ecomail.fr> robinechuca <61911191+robinechuca@users.noreply.github.com>
 Robin Richard <raisin@ecomail.fr> Nornort <francois.lalleve@laposte.net>
-Ayush Kumar <ayushk7102@gmail.com> Ayush Kumar <65803868+ayushk7102@users.noreply.github.com>
```
Comment (oscarbenjamin) [inline .mailmap:570]: Why is this deleted?

Comment (praneethratna) [reply .mailmap:570]: I'm not sure but i guess that was done after rebasing.

Comment (oscarbenjamin) [reply .mailmap:570]: I guess it is not needed because the 65803868 email address does not seem to be in the git log.

File: sympy/simplify/fu.py [hunk 3]
```diff
@@ -1645,6 +1646,11 @@ def fu(rv, measure=lambda x: (L(x), x.count_ops())):
     rv = sympify(rv)
     if not isinstance(rv, Expr):
         return rv.func(*[fu(a, measure=measure) for a in rv.args])
+    else:
+        n, d = fraction(rv)
+        if d.is_zero is None:
+            if simplify(d) is S.Zero:
+                return nan
```
[the remark this answers is not in the document]
Comment (praneethratna) [reply sympy/simplify/fu.py:1653]: I used it to simplify `sin(x)**2 + cos(x)**2 - 1` to 0 , I guess there must be some other method to do that without using `simplify` which I'm not aware of.

File: .mailmap (hunk 2 above)
Comment (oscargus) [reply .mailmap:570]: I checked the commit log and it seems like there is no "GitHub account" associated with some recent commits by Ayush Kumar, i.e., it is not possible to click the name for his commits. Probably has nothing to do with this, but since it is the same person I thought it was worth mentioning here.

File: sympy/simplify/fu.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (oscargus) [reply sympy/simplify/fu.py:1653]: `TR5` (or `TR6`) will detect this particular case.

An intermediate approach may be to call `futrig(d)` here which may cover a few more cases. But hard to say if it is worth it in general. If the aim is to avoid cancelling 0/0 it probably makes sense rather than trying to add one special case at a time.
